<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Serial_Test_Widget</class>
 <widget class="QWidget" name="Serial_Test_Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>650</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>650</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Serial Test Tool</string>
  </property>
  <widget class="QGroupBox" name="send_serial_info_gb_">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>40</y>
     <width>781</width>
     <height>51</height>
    </rect>
   </property>
   <property name="title">
    <string>Send Data Serial Port Info</string>
   </property>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>21</y>
      <width>751</width>
      <height>22</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="send_serial_port_lbl_">
       <property name="text">
        <string>Port</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="send_serial_port_cb_">
       <property name="editable">
        <bool>true</bool>
       </property>
       <item>
        <property name="text">
         <string>COM1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>COM2</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>/dev/ttyAMA0</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>/dev/ttya</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>/dev/ttyb</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>COM3</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>COM4</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>COM5</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>COM6</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>COM7</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>COM8</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>COM9</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>COM10</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="send_baud_lbl_">
       <property name="text">
        <string>Baud</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="send_baud_cb_">
       <property name="editable">
        <bool>true</bool>
       </property>
       <item>
        <property name="text">
         <string>9600</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>19200</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>38400</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>57600</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>115200</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>921600</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="send_data_bits_lbl_">
       <property name="text">
        <string>DataBits</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="send_data_bits_cb_">
       <item>
        <property name="text">
         <string>8 bits</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>7 bits</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="send_parity_lbl_">
       <property name="text">
        <string>Parity</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="send_parity_cb_">
       <item>
        <property name="text">
         <string>None</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Even</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Odd</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="send_stop_bits_lbl_">
       <property name="text">
        <string>Stop Bits</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="send_stop_bits_cb_">
       <item>
        <property name="text">
         <string>1 bit</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>1.5 bits</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>2 bits</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="send_flow_ctrl_lbl_">
       <property name="text">
        <string>Flow Control</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="send_flow_ctrl_cb_">
       <property name="editable">
        <bool>true</bool>
       </property>
       <item>
        <property name="text">
         <string>none</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>hardware</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>software</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>781</width>
     <height>25</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="conn_lbl_">
        <property name="text">
         <string>Connection</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="conn_cb_">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <item>
         <property name="text">
          <string>Serial Port</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QGroupBox" name="send_data_gb_">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>140</y>
     <width>781</width>
     <height>141</height>
    </rect>
   </property>
   <property name="title">
    <string>Send Data</string>
   </property>
   <widget class="QTextEdit" name="send_data_txt_">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>761</width>
      <height>111</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="recv_data_gb_">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>370</y>
     <width>781</width>
     <height>271</height>
    </rect>
   </property>
   <property name="title">
    <string>Receive Data</string>
   </property>
   <widget class="QTextEdit" name="recv_data_txt_">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>761</width>
      <height>241</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>110</y>
     <width>781</width>
     <height>25</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_4">
    <item>
     <widget class="QLabel" name="send_data_type_lbl_">
      <property name="text">
       <string>Send Data Type</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QComboBox" name="send_data_type_cb_">
      <property name="editable">
       <bool>false</bool>
      </property>
      <item>
       <property name="text">
        <string>HEX</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>ASCII</string>
       </property>
      </item>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="send_interval_lbl_">
      <property name="text">
       <string>Send Interval(ms)</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QComboBox" name="send_interval_cb_">
      <property name="editable">
       <bool>true</bool>
      </property>
      <property name="currentText" stdset="0">
       <string notr="true">100</string>
      </property>
      <item>
       <property name="text">
        <string>100</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>200</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>500</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>1000</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>2000</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>5000</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>10000</string>
       </property>
      </item>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="send_repeatedly_cb_">
      <property name="text">
       <string>Send Repeatedly</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="send_btn_">
      <property name="text">
       <string>Send</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QGroupBox" name="recv_serial_info_gb_">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>280</y>
     <width>781</width>
     <height>51</height>
    </rect>
   </property>
   <property name="title">
    <string>Receive Data Serial Port Info</string>
   </property>
   <widget class="QWidget" name="layoutWidget_3">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>21</y>
      <width>751</width>
      <height>22</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_6">
     <item>
      <widget class="QLabel" name="recv_serial_port_lbl_">
       <property name="text">
        <string>Port</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="recv_serial_port_cb_">
       <property name="editable">
        <bool>true</bool>
       </property>
       <item>
        <property name="text">
         <string>COM1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>COM2</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>/dev/ttyAMA0</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>/dev/ttya</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>/dev/ttyb</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>COM3</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>COM4</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>COM5</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>COM6</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>COM7</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>COM8</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>COM9</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>COM10</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="recv_baud_lbl_">
       <property name="text">
        <string>Baud</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="recv_baud_cb_">
       <property name="editable">
        <bool>true</bool>
       </property>
       <item>
        <property name="text">
         <string>9600</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>19200</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>38400</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>57600</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>115200</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>921600</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="recv_data_bits_lbl_">
       <property name="text">
        <string>DataBits</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="recv_data_bits_cb_">
       <item>
        <property name="text">
         <string>8 bits</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>7 bits</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="recv_parity_lbl_">
       <property name="text">
        <string>Parity</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="recv_parity_cb_">
       <item>
        <property name="text">
         <string>None</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Even</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Odd</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="recv_stop_bits_lbl_">
       <property name="text">
        <string>Stop Bits</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="recv_stop_bits_cb_">
       <item>
        <property name="text">
         <string>1 bit</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>1.5 bits</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>2 bits</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="recv_flow_ctrl_lbl_">
       <property name="text">
        <string>Flow Control</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="recv_flow_ctrl_cb_">
       <property name="editable">
        <bool>true</bool>
       </property>
       <item>
        <property name="text">
         <string>none</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>hardware</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>software</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QWidget" name="layoutWidget_4">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>340</y>
     <width>781</width>
     <height>25</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_7">
    <item>
     <widget class="QPushButton" name="recv_btn_">
      <property name="text">
       <string>Receive</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="stop_recv_btn_">
      <property name="text">
       <string>Stop</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
